src/main/java/com/iota/iri/service/API.java

```diff
@@ -1014,11 +1366,10 @@ private AbstractResponse getBalancesStatement(final List<String> addresses, fina
                 balances.put(address, value);
             }
 
-            final Set<Hash> visitedHashes;
-            final Map<Hash, Long> diff;
+            final Set<Hash> visitedHashes = new HashSet<>();
+            final Map<Hash, Long> diff = new HashMap<>();
 
-            visitedHashes = new HashSet<>();
-            diff = new HashMap<>();
+            // Then check if all tips are present and we can update the transactions then through
```

Comment:
then through? :o through wut

Reply:
Oh yea, i wasnt sure how this worked!
@GalRogozinski Could you help me out here with instance.ledgerValidator.updateDiff? :)